### PR TITLE
PMPRO_IPN_DEBUG logic refactoring

### PR DESCRIPTION
**Pull request by mircobabini**
### All Submissions:

* [x] Have you followed the [Contributing guideline](CONTRIBUTING.MD)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

- Some refactoring;
- Logic fix: should avoid counterintuitive interpretation of false;
- Filter on ipn_logfile: I want to keep the history, can't log into pmpro folder;

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: services/ipnhandler.php [hunk 1]
```diff
@@ -361,22 +361,27 @@ function pmpro_ipnExit() {
...
+			} elseif ( PMPRO_IPN_DEBUG === "log" ) {
+				//file
+				$logfile = apply_filters( 'pmpro_ipn_logfile', dirname( __FILE__ ) . "/../logs/ipn.txt" );
+				$loghandle = fopen( $logfile, "a+" );
+				fwrite( $loghandle, $logstr );
+				fclose( $loghandle );
+			} elseif ( is_email( PMPRO_IPN_DEBUG ) ) {
+				//email to specified address
+				wp_mail( PMPRO_IPN_DEBUG, get_option( "blogname" ) . " IPN Log", nl2br( $logstr ) );							
+			} elseif ( !!PMPRO_IPN_DEBUG ){
```
Comment (ideadude) [inline services/ipnhandler.php:381]: Why !!PMPRO_IPN_DEBUG instead of just else {} or elseif ( PMPRO_IPN_DEBUG === true )?

Comment (mircobabini) [reply services/ipnhandler.php:381]: Well, I think "PMPRO_IPN_DEBUG === true" can be ok. Actually !!PMPRO_IPN_DEBUG is a cast to boolean so it's the same thing.

Comment (ideadude) [reply services/ipnhandler.php:381]: I like using an else {}

PMPRO_IPN_DEBUG is defined, it's not 'log' or an email address, let's assume it's true or 1 or yes and people meant for an email to be sent.

Can you update? Then I'll merge in.

Comment (mircobabini) [reply services/ipnhandler.php]: Done!

**Pull request merged**